src/metabase/api/field.clj

```diff
@@ -65,16 +100,49 @@
      [:distincts (metadata/field-distinct-count field)]]))
 
 (def ^:private empty-field-values
-  {:values {} :human_readable_values {}})
+  {:values []})
 
 (api/defendpoint GET "/:id/values"
   "If `Field`'s special type derives from `type/Category`, or its base type is `type/Boolean`, return
    all distinct values of the field, and a map of human-readable values defined by the user."
   [id]
   (let [field (api/read-check Field id)]
-    (if-not (field-should-have-field-values? field)
-      empty-field-values
-      (create-field-values-if-needed! field))))
+    (if-let [field-values (and (field-should-have-field-values? field)
+                               (create-field-values-if-needed! field))]
+      (-> field-values
+          (assoc :values (field-values->pairs field-values))
+          (dissoc :human_readable_values))
+      {:values []})))
+
+(api/defendpoint POST "/:id/dimension"
+  "Sets the dimension for the given field at ID"
+  [id :as {{dimension-type :type dimension-name :name human_readable_field_id :human_readable_field_id} :body}]
```

Comment:
There's a caveat when doing map destructuring that renames keys for API endpoints: the API documentation generation code looks at the values in the param validation schema map and uses that to generate the endpoint description. 

Try it for yourself -- run `lein run api-documentation` -- and you'll see that the documentation will tell people they need to supply `dimension-type` (even though the actual name of the param is `type`). However avoiding renaming (e.g. a simple `:keys` destructure) would leave us with the dilemma of `name` and `type` overshadowing the `clojure.core` fns.

I think there's two ways we can have our cake and eat it too here:

1.  We could just rename these API parameters to things like `dimension_type` and `dimension_name` so the renaming during destructuring isn't needed. However this is overly verbose and I don't like naming things certain names just to make API documentation code happy
2. (Better) we can add a new wrapper function that allows you to specify an "actual" param name. Something like:

    ```clojure
    {dimension-type (su/api-key "type" (s/enum "internal" "external"))}
    ```

    Or perhaps

    ```clojure
    {(su/api-key "type" dimension-type) (s/enum "internal" "external")}
    ```

    We already have something similar with the `with-api-error-message` in the schema utils namespace, so this would largely be a matter of copying this pattern and tweaking the functions used by the `defendpoint` macro for generating documentation.
 
    This would have the added bonus of letting us do renaming destructures in API endpoints more often which would let us keep more things in `lisp-case` and make the code prettier lMO.

All that being said, this PR is already sort of huge so tweaks to the defendpoint macro would probably better be done as a follow-on so this PR doesn't get too huge. I think maybe the best option is to open an issue to remind ourselves to look at it in the future. I can do so if you want, I already have a pretty idea how to get it working nicely